File: Classes/ThemeObject/CAGradientLayer+LWTheme.m
//
//  CAGradientLayer+LWTheme.m
//  LWThemeObjC
//
//  Created by lalawue on 2019/12/22.
//

#import "CAGradientLayer+LWTheme.h"
#import "LWTheme_Tools.h"

@implementation CAGradientLayer (LWTheme)

- (NSArray<NSString *> *)lw_colors
{
    return (NSArray<NSString *> *)[self lw_valueForSetter:@selector(setLw_colors:)];
}

- (void)setLw_colors:(NSArray<NSString *> *)lw_colors
{
    if ([lw_colors isKindOfClass:[NSArray class]] && lw_colors.count) {
        NSMutableArray *array = [NSMutableArray new];
        for (NSString *colorPattern in lw_colors) {
            UIColor *colorValue = [self lw_colorForPattern:colorPattern];
            if (colorValue) {
                [array addObject:(id)colorValue.CGColor];
            }
        }
        self.colors = array;
        [self lw_storeValue:lw_colors forSetter:@selector(setLw_colors:)];
    }
}

@end

File: Classes/ThemeObject/CAGradientLayer+LWTheme.swift
//
//  GradientLayer+LWTheme.swift
//  LWThemeSwift
//
//  Created by lalawue on 2019/12/22.
//

import UIKit

extension CAGradientLayer {
    
    @objc public var lw_colors : [String]? {
        
        get {
            return lw_valueForSetter(#selector(setter: CAGradientLayer.lw_colors)) as? [String]
        }
        
        set (colorPatterns) {
            if ((colorPatterns == nil) || (colorPatterns!.count <= 0)) {
                return
            }
            var colors : [Any] = []
            for pattern in colorPatterns! {
                let colorValue = lw_colorForPattern(pattern)
                if (colorValue != nil) {
                    colors.append(colorValue!.cgColor)
                }
            }
            self.colors = colors
            lw_storeValue(colorPatterns, setter: #selector(setter: CAGradientLayer.lw_colors))
        }
    }

}
